scripts/compare.coffee

```diff
@@ -1,28 +1,28 @@
 # Description:
-#  Returns a github release/branch comparison URL for a
+#  Generates a GitHub comparison URL
 #
 # Configuration:
-#   HUBOT_GITHUB_USER
+#   HUBOT_GITHUB_USER - GitHub org/user to use if none explicitly given
 #
 # Commands:
-#   hubot compare me <repo> <release/branch> <release/branch> - Reply with url to github compare ¬
+#   hubot compare me <repo> <treeish> <treeish> - Reply with url to github compare page
+#
+# Notes:
+#   None
 #
 # Author:
-#  markhuge 
+#  markhuge
 
 user = process.env.HUBOT_GITHUB_USER
+
 module.exports = (robot) ->
-  if user
-    robot.respond ///
-        compare\ me\ (:?#{user}\/)*([\w-]+)\ ([\w\.\/\\-]+)\s([\w\.\/\\-]+)
-      ////i, (msg) ->
-      msg.send "https://github.com/#{user}/#{msg.match[2]}/compare/#{msg.match[3]}...#{msg.match[4]}"
+  robot.respond /compare (?:me )?([\w-.\/]+) ([\w-.\/]+) (?:to )?([\w-.\/]+)/i, (msg) ->
+    [_, repo, from, to] = msg.match
```

Comment:
Hehe, thanks! Probably shouldn't take the credit though, my brain probably pinched it from something it saw in the ocean of GitHub! :grinning: